SHS-5629: Editors can more easily edit a caption

codechefmarc opened the pull request:
# READY FOR REVIEW

## Summary
- Adds a patch to Drupal core to add an edit icon next to media items that use the core media library form display.
- This allows for editing of the media item in a modal while in a node edit form.
- Tugboat testing instance: https://pr1580-lqaarerjbqqcvxksaxnggcqcjx0lz4qh.tugboatqa.com

## Need Review By (Date)
7/22/2024

## Urgency
low

## Steps to Test
1. Login to the site 
2. Edit the homepage or a page that has a "Banner image(s) with partial overlay and text

![Screenshot 2024-07-19 at 3 48 55 PM](https://github.com/user-attachments/assets/3c0c0816-ab15-45a5-9d7a-64720921e915)

3. Click "Edit" in the "Banner image(s) with partial overlay and text" paragraph

![Screenshot 2024-07-19 at 3 49 07 PM](https://github.com/user-attachments/assets/ea3db652-456a-4e24-bc19-dc5c67482e7a)

4. Click "Edit" on one of the "Banner image with overlay and text" paragraphs

![Screenshot 2024-07-19 at 3 49 16 PM](https://github.com/user-attachments/assets/c358a293-78f1-4bd3-826c-9a21596d709f)

5. Verify that there's now a pencil icon in addition to the "X" icon on top of the Image media

![Screenshot 2024-07-19 at 3 49 22 PM](https://github.com/user-attachments/assets/a9efe79e-1b02-4a39-8a68-83f0e438881f)

6. Click the pencil icon
7. Verify that a modal pops up that has the media edit form for that image

![Screenshot 2024-07-19 at 3 49 29 PM](https://github.com/user-attachments/assets/0ee87245-a3a4-4bbe-a72d-f337e87bb440)

8. Change the text in the "Caption/Credit" text area
9. Save the media in the modal
10. Save the page
11. Verify that the caption text on the slide that was edited has now changed

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207858777652180

Commit:
feat(SHS-5629): Add patch to allow editing of media items in a modal from a node edit page

Commit:
fix(SHS-5629): Remove todo in patch file

codechefmarc committed:
SHS-5661: Remove legacy fields (#1577)

* feat(SHS-5661): Remove paragraph field Overlay Color on banner image with text box

* feat(SHS-5661): Remove other references to the overlay field

* feat(SHS-5661): Add update hook to remove legacy field

* fix(SHS-5661): Fix linting error

* fix(shs-5661): remove hero-text-overlay variants and update viewmode config

---------

Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

mariannuar committed:
SHS-4929: Hide caption/credits on images in default views (#1578)

* feat(shs-4929): add new media view mode and update news default views

* feat(shs-4929):  update people default views

* feat(shs-4929):  update publications default views

---------

Co-authored-by: Mari Nez <mariannuar@gmail.com>
Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

cienvaras committed:
Merge branch '11.1.4-release' of github.com:SU-HSDO/suhumsci into fk-stnfd-sprint-55

cienvaras commented on composer.json:
```diff
@@ -240,7 +240,8 @@
                 "CKEditor 5 headings plugin paragraph label change": "patches/ckeditor5-paragraph-rename.patch",
-                "Allow hs group blocks in layout_builder": "patches/layout_builder-allow-hs-blocks.patch"
+                "Allow hs group blocks in layout_builder": "patches/layout_builder-allow-hs-blocks.patch",
+                "Allow media items to be edited in a modal when using the field widget https://www.drupal.org/project/drupal/issues/2985168": "patches/2985168-172.patch"
```
@codechefmarc Can you change this to pull the patch directly from drupal.org instead of downloading it and using a local copy, please? That's the standard practice in this project, we only use local patches if it's something that's not in the issue queue. Thanks!

codechefmarc replied:
Fixed!

cienvaras reviewed (changes requested):
@codechefmarc I can see the edit button and open the modal, but I found an issue with the modal position. It's not always centered, and sometimes the "Save" button cannot be accessed:
<img width="1477" alt="Screenshot 2024-07-24 at 9 45 49 PM" src="https://github.com/user-attachments/assets/f260e63c-5b17-4e65-9aaa-46a3cae18946">

Not sure what's the cause of the issue, because every other time the modal is rendered in the right position.

Also, please check the other comment I left in the code. Thanks!

Commit:
refactor(SHS-5629): Switch local patch for remote patch, add patch to fix missing save button on media modal

Commit:
Merge branch 'develop' into SHS-5629--media-caption-edit

codechefmarc commented:
@cienvaras - Ok, switched out the patch for a d.o. version and also added a patch that I think fixes the save button issue. It had to do with Gin, I believe. I had a hard time recreating as it worked for me without that patch, but once I put that patch in, it works all the time, I think? Please re-test. What browser were you using? I tested on Arc and Safari.

codechefmarc committed:
SHS-5675: Helping users avoid 'Table cell missing context (e.g. headings) A11y warning (#1579)

* feat(SHS-5675): Add default table header rows to new tables in CKEditor

* fix(SHS-5675): Fix linting error

* docs(SHS-5675): Add comments describing change and source of change.

* feat(SHS-5675): Update table heading color in CKEditor

* fix(shs-5675): change color variable for table headings

---------

Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

cienvaras committed:
Merge branch 'fk-stnfd-sprint-55' of github.com:SU-HSDO/suhumsci into SHS-5629--media-caption-edit

cienvaras committed:
Merge branch '11.1.4-release' of github.com:SU-HSDO/suhumsci into fk-stnfd-sprint-55

cienvaras committed:
Merge branch 'fk-stnfd-sprint-55' of github.com:SU-HSDO/suhumsci into SHS-5629--media-caption-edit

cienvaras committed:
fix(shs-5629): revert composer.lock changes

cienvaras commented:
@codechefmarc I initially tested in Arc in Mac. After applying the gin patch, I tested again in Arc, Chrome and Safari and I still have the issue in Arc and Chrome, Safari works well.

It seems that the issue is related to scrolling. In most cases, the dialog is initially shown in the right position, but if I close it, scroll the page and open it again it's displayed too low on the page.

I'll try to get someone else to test this and discard that the problem is happening only on my computer, but I'll hold on merging this until we're sure that the issue is solved.

ahughes3 commented:
> It seems that the issue is related to scrolling. In most cases, the dialog is initially shown in the right position, but if I close it, scroll the page and open it again it's displayed too low on the page.
> 
> I'll try to get someone else to test this and discard that the problem is happening only on my computer, but I'll hold on merging this until we're sure that the issue is solved.

@codechefmarc @cienvaras  I tested today using Chrome on a Mac and the issue is still there. I tested previously before the patch was added and experienced the same issue. The modal opens up but I can't scroll to access the save button

https://www.loom.com/share/cef1a182ef7745c1988cb31241851c8e?sid=ee07a8e6-f092-4577-812b-c98fefe0eee9

codechefmarc commented:
@ahughes3 , @cienvaras  - Thank you for testing, I'll get back on this ticket and research how we can fix that odd bug. I'm sure there is a way.

Commit:
fix(SHS-5629): Possible fix for dialog box too low on screen

codechefmarc commented:
@ahughes3 and @cienvaras  - I pushed some changes up and tested the dialog after scrolling and it appear to be working for me, please re-test. Thank you!
https://pr1580-lqaarerjbqqcvxksaxnggcqcjx0lz4qh.tugboatqa.com/

Commit:
fix(SHS-5629): Fix dialog too low on screen issue (clear browser cache to fix)

cienvaras committed:
Merge branch 'fk-stnfd-sprint-56' of github.com:SU-HSDO/suhumsci into SHS-5629--media-caption-edit

cienvaras committed:
Merge branch 'fk-stnfd-sprint-56' of github.com:SU-HSDO/suhumsci into SHS-5629--media-caption-edit

cienvaras committed:
fix(shs-5629): remove unnecesary change

cienvaras committed:
fix(shs-5629): update patch
